src/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab.tsx

```diff
@@ -1,94 +1,339 @@
-import React, { FC } from "react";
+import React, { FC, useState } from "react";
 
 import Box from "@/components/Box/Box";
 import Flex from "@/components/Flex/Flex";
 import P, { Heading } from "@/components/Typography";
 import Card from "@/components/Card/Card";
-import DropdownSelect from "@/components/DropdownSelect/DropdownSelect";
 import { CurriculumUnitsTabData } from "@/node-lib/curriculum-api-2023";
+import Icon from "@/components/Icon/Icon";
+import OutlineHeading from "@/components/OutlineHeading/OutlineHeading";
+import OakLink from "@/components/OakLink/OakLink";
+import Button from "@/components/Button/Button";
 
 type UnitsTabProps = {
   data: CurriculumUnitsTabData;
 };
 
-const SequenceTab: FC<UnitsTabProps> = ({ data }) => {
-  const { units, threads } = data;
+const UnitsTab: FC<UnitsTabProps> = ({ data }) => {
+  type Unit = CurriculumUnitsTabData["units"][number];
+
+  interface Subject {
+    subject: string;
+    subject_slug: string;
+  }
+
+  interface Domain {
+    title: string;
+    tag_id: number | null;
+  }
+
+  interface Tier {
+    tier: string;
+    tier_slug: string;
+  }
+
+  interface YearSelection {
+    [key: string]: {
+      subject?: Subject | null;
+      domain?: Domain | null;
+      tier?: Tier | null;
+    };
+  }
+
+  const yearData: {
+    [key: string]: {
+      units: Unit[];
+      childSubjects: Subject[];
+      domains: Domain[];
+      tiers: Tier[];
+    };
+  } = {};
+
+  data.units.forEach((unit) => {
```

Comment:
Seems like there's a fair amount of wrangling with data inline in this component -- would be great to the functionality it tested ... could pull it out into its own function?